### PR TITLE
Moved configuration into network preset instead of constants.

### PR DESCRIPTION
**Pull request by rolfyone**
Now that `MAX_CHUNK_SIZE` and `GOSSIP_MAX_SIZE` are in configuration, we no longer need separate constants to represent them in the spec when they change in Bellatrix.

I've changed the usage, and put the values into the presets, but I'm not sure if I've updated the descriptions in the best way...

This is following on from the work in #3375 where a number of constants got moved into configuration, so we no longer need these constants to be separately represented, they can simply be updated in presets.

### CODE REVIEW

**Review by dapplion — APPROVED** [review]

File: presets/minimal/bellatrix.yaml [hunk 1]
```diff
@@ -19,3 +19,11 @@ MAX_TRANSACTIONS_PER_PAYLOAD: 1048576
...
 # 2**5 (= 32)
 MAX_EXTRA_DATA_BYTES: 32
+
+# Networking
+# ---------------------------------------------------------------
+# `10 * 2**20` (= 10485760, 10 MiB)
+GOSSIP_MAX_SIZE: 10485760
+
+# `10 * 2**20` (= 10,485,760, 10 MiB)
+MAX_CHUNK_SIZE: 1048576
```
Comment (hwwhww) [inline presets/minimal/bellatrix.yaml:29]:
```suggestion
MAX_CHUNK_SIZE: 10485760

```

File: presets/mainnet/bellatrix.yaml [hunk 2]
```diff
@@ -19,3 +19,11 @@ MAX_TRANSACTIONS_PER_PAYLOAD: 1048576
...
 # 2**5 (= 32)
 MAX_EXTRA_DATA_BYTES: 32
+
+# Networking
+# ---------------------------------------------------------------
+# `10 * 2**20` (= 10485760, 10 MiB)
+GOSSIP_MAX_SIZE: 10485760
+
+# `10 * 2**20` (= 10,485,760, 10 MiB)
+MAX_CHUNK_SIZE: 1048576
```
Comment (hwwhww) [inline presets/mainnet/bellatrix.yaml:29]:
```suggestion
MAX_CHUNK_SIZE: 10485760
```

**Review by hwwhww — COMMENTED** [review]
Thank you @rolfyone for catching it!

However, it seems odd to have them concurrently in both the preset and the config. An alternative solution could be to follow the pattern of other preset updates like `INACTIVITY_PENALTY_QUOTIENT_BELLATRIX`. We could keep the suffix and relocate them to either the preset or the config.

**Comment by rolfyone** [comment]
I agree, I hadn't really taken much time to look at the formats of the other files.  Updated that, and it looks like we don't need to call out changes in config now that they're preset so removed the updated descriptions.
I had copied and pasted so many times those values, sorry I got the chunk sizes wrong!

**Comment by djrtwo** [comment]
It's not clear to me that you can get rid of the individual configuration values at the time of forks, otherwise you might downscore a node that has already upgraded pre-fork (or vice versa)

**Comment by rolfyone** [comment]
> It's not clear to me that you can get rid of the individual configuration values at the time of forks, otherwise you might downscore a node that has already upgraded pre-fork (or vice versa)

I'm not sure what you mean by this specifically...

We haven't got rid of the value, just that it's now represented in the presets... When making the changes in teku I realised that we have this bellatrix constant that should really just be part of the spec presets that we're reading, otherwise bellatrix will forever have to have special code on it's gossip size and chunk size...

This should be a logical equivalence to the 2 constants, just that now the value being altered is represented in the presets...

Because it's preset, and the description of the usage of those is identical to phase 0, it seems that there's no point repeating it unless we're calling out that it's now 10MiB, and we didnt seem to do that for other values that I'd looked at in the presets on the beacon-chain doc...

I'm happy to put the comments back about the values increasing from 1MiB to 10Mib if that's clearer?

**Comment by djrtwo** [comment]
Ah, you're right. I wasn't paying attention to the distinction that presets are per fork. sorry!

**Comment by djrtwo** [comment]
But yeah, leaving a call-out about the increase and/or how/when to do the change would probably be good

**Comment by hwwhww** [comment]
@rolfyone 

Do clients use 10 MiB as the maximum size for gossiping the phase0 objects, if any?
If no client uses 1 MiB anymore, can we just update the phase0 setting?

---

Separately, my proposal of relocating them to preset would be like this: https://github.com/ethereum/consensus-specs/commit/822b7440dd0a214e18c82fd10c9a9bb0e09baf5f

Rationales:
- We don't mix preset and config in the spec
- We can't use duplicate names in the preset to present the "preset update" without some hack.
- To follow the same rule of beacon chain preset updates. Although I understand they may not exactly the same. 😬

**Comment by rolfyone** [comment]
Oh! Ok I've never done presets, maybe 3rd time lucky, I think I had the same problem in code that you're describing, where i can't just update the value, which I had assumed we could do :/

Thanks for your patience... I'll get to this this morning.

**Comment by rolfyone** [comment]
> Do clients use 10 MiB as the maximum size for gossiping the phase0 objects, if any?
> If no client uses 1 MiB anymore, can we just update the phase0 setting?

potentially yes we could just update that setting... We were just using 10MB as soon as Bellatrix was defined....

**Comment by tersec** [comment]
> > Do clients use 10 MiB as the maximum size for gossiping the phase0 objects, if any?
> > If no client uses 1 MiB anymore, can we just update the phase0 setting?
> 
> potentially yes we could just update that setting... We were just using 10MB as soon as Bellatrix was defined....

Nimbus also simply always uses 10MB message size limits now, so would prefer to update the phase0 setting too.

**Comment by rolfyone** [comment]
simplified based on feedback. ready to look at again.

**Review by Heheehd — COMMENTED** [review]
H

**Pull request merged**